src/safeposix/syscalls/net_calls.rs

```diff
@@ -3123,69 +3301,87 @@ impl Cage {
                         continue;
                     }
 
+                    // get the events to monitor
                     let events = value.events;
                     let mut structpoll = PollStruct {
                         fd: key,
                         events: 0,
                         revents: 0,
                     };
+                    // check for each supported event
                     if events & EPOLLIN as u32 > 0 {
                         structpoll.events |= POLLIN;
                     }
                     if events & EPOLLOUT as u32 > 0 {
                         structpoll.events |= POLLOUT;
                     }
-                    if events & EPOLLERR as u32 > 0 {
-                        structpoll.events |= POLLERR;
+                    if events & EPOLLPRI as u32 > 0 {
```

Comment:
Why are we changing this flag to `EPOLLPRI`.

Reply:
This is a bug, there is some misunderstanding on this flag (EPOLLERR) before. This flag is supposed to be the flag used for monitoring exception conditions of file description, but it is actually the "EPOLLPRI" flag marks for exception condition, and EPOLLERR is the flag that would only be set in return event.